.github/workflows/ci.yml

```diff
@@ -0,0 +1,92 @@
+name: Run checks on pull requests and pushes on main branch
+
+on:
+  pull_request: {}
+  push:
+    branches: [main]
+
+jobs:
+  tests:
+    name: Checks of unit tests and format checks
+    runs-on: "${{ matrix.os }}"
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      fail-fast: false
```

Comment:
Would setting [this to true](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions) be closer to our usual paradigm?

Reply:
yeah the behavior is to cancel other parallel jobs, in this case it would be different python versions, and windows mac system jobs. (guess it's a bit different than our existing jenkins job which is just one time run with a single environment)
I feel this would be helpful to be false to directly know which jobs are falling when there are multiple ones failed. For example, if there are 3 jobs failing out of total 9 jobs. we can know it directly instead of seeing 1 failed, try to fix that one. And potentially on the next try, the other job failed first so we couldn't know if the one we are trying to fix works or not.
But yeah if we still think fail first is what we want, i can change

Reply:
Your comment makes sense and I now agree with `fail-fast: false`.